### ISSUE TITLE
Support for gCarvin firmware

### ISSUE THREAD

**Issue by cheton** [cncjs/cncjs#203]
### Description

The [gCarvin firmware](https://github.com/inventables/gCarvin) is a Grbl fork which is intended for use on the Carvey 3D carving machine from Inventables. However, the gCarvin firmware cannot work on CNCjs 1.9.x, since the welcome message is different from Grbl, and CNCjs cannot recognize the message so that it won't turn on the ready flag. Let's see some examples:

**Grbl**
```
Grbl 0.9f ['$' for help]
Grbl 1.1f ['$' for help]
```

**gCarvin**
```
gCarvin 2.0.0 ['$' for help]
```

### Proposal

Do not use a hard-coded firmware name in the regular expression.

Before
```
/^Grbl\s*(\d+\.\d+[a-zA-Z]?)/
```

After
```
/^([a-zA-Z0-9]+)\s+((?:\d+\.){1,2}\d+[a-zA-Z0-9\-\.]*)\s+(\[[^\]]+\])/
```

### Test Cases
* Grbl 0.9f ['$' for help]
* Grbl 1.1f ['$' for help]
* gCarvin 2.0.0 ['$' for help]

**Comment by cheton** [comment]
```js
const message = 'gCarvin 2.0.0 ['$' for help]';
message.match(/^([a-zA-Z0-9]+)\s+((?:\d+\.){1,2}\d+[a-zA-Z0-9\-\.]*)\s+(\[[^\]]+\])/);
// => ["gCarvin 2.0.0 ['$' for help]", "gCarvin", "2.0.0", "['$' for help]", index: 0, input: "gCarvin 2.0.0 ['$' for help]"]
```

**Comment by cheton** [comment]
Fixed in 1.9.10
